src/main/scala/com/advancedtelematic/director/data/DataType.scala

```diff
@@ -57,7 +57,14 @@ object DbDataType {
                        sha256: SHA256Checksum,
                        uri: Option[Uri])
 
-  case class Assignment(ns: Namespace, deviceId: DeviceId, ecuId: EcuIdentifier, ecuTargetId: EcuTargetId, correlationId: CorrelationId, inFlight: Boolean)
+  case class Assignment(ns: Namespace, deviceId: DeviceId, ecuId: EcuIdentifier, ecuTargetId: EcuTargetId,
+                        correlationId: CorrelationId, inFlight: Boolean) {
+    def toProcessedAssignment(canceled: Boolean = false): ProcessedAssignment =
```

Comment:
Not sure this should have a default `canceled` this might introduce some errors if we use this wrong. 

The only place we use this method in production code is when transforming the assignment into a processed assignment. I would prefer to this explicitly there rather than have this method that can be overused... In tests we can just compare explicitly all attributes...

Reply:
you mean like adding a method like `toProcessedAssignmentCancelled`?

Reply:
Ah you mean copy it and setting the flag?

Reply:
I meant really creating `ProcessedAssignment(...)` rather than having a method for it. Not sure if it's better.

Reply:
Ah, right. I left the method in since both times it is used in production code is in a map.